e2etest/bgptests/bgp.go

```diff
@@ -600,6 +604,190 @@ var _ = ginkgo.Describe("BGP", func() {
+		}
+
+		ginkgo.By("Creating the service advertised to all peers")
+		resources := config.Resources{
+			Pools: []metallbv1beta1.IPAddressPool{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "bgp-test",
+					},
+					Spec: metallbv1beta1.IPAddressPoolSpec{
+						Addresses: poolAddresses,
+					},
+				},
+			},
+			Peers:   peers,
+			BGPAdvs: []metallbv1beta1.BGPAdvertisement{bgpAdv},
+		}
+
+		err := ConfigUpdater.Update(resources)
+		Expect(err).NotTo(HaveOccurred())
+
+		svc, _ := testservice.CreateWithBackend(cs, testNamespace, "external-local-lb", func(svc *corev1.Service) {
+			testservice.TrafficPolicyCluster(svc)
+			tweak(svc)
+		})
+		svcDeleted := false
+		defer func() {
+			if !svcDeleted {
+				testservice.Delete(cs, svc)
+			}
+		}()
+
+		for _, i := range svc.Status.LoadBalancer.Ingress {
+			ginkgo.By("validate LoadBalancer IP is in the AddressPool range")
+			ingressIP := jigservice.GetIngressPoint(&i)
+			err = config.ValidateIPInRange(resources.Pools, ingressIP)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		nodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		nodesNames := []string{}
+		for _, n := range nodes.Items {
+			nodesNames = append(nodesNames, n.Name)
+		}
+
+		ginkgo.By("Verifying all nodes create a status for the service")
+		Eventually(func() error {
+			for _, n := range nodesNames {
+				s, err := status.ForBGPServiceAndNode(ConfigUpdater.Client(), svc, n)
+				if err != nil {
+					return err
+				}
+
+				if !reflect.DeepEqual(peersNames, s.Status.Peers) {
+					return fmt.Errorf("expected status peers to be %v, got %v for node %s", peersNames, s.Status.Peers, n)
+				}
+			}
+
+			return nil
+		}, 2*time.Minute, 2*time.Second).ShouldNot(HaveOccurred())
+
+		ginkgo.By("Updating the peers of the adv to all peers and a nonexistent one the statuses should remain the same as before")
+		err = ConfigUpdater.Client().Get(context.TODO(), types.NamespacedName{Namespace: bgpAdv.Namespace, Name: bgpAdv.Name}, &bgpAdv)
+		Expect(err).ToNot(HaveOccurred())
+		bgpAdv.Spec.Peers = append(peersNames, "dummy")
+		err = ConfigUpdater.Client().Update(context.TODO(), &bgpAdv)
+		Expect(err).ToNot(HaveOccurred())
+
+		Consistently(func() error {
+			for _, n := range nodesNames {
+				s, err := status.ForBGPServiceAndNode(ConfigUpdater.Client(), svc, n)
+				if err != nil {
+					return err
+				}
+
+				if !reflect.DeepEqual(peersNames, s.Status.Peers) {
+					return fmt.Errorf("expected status peers to be %v, got %v for node %s", peersNames, s.Status.Peers, n)
+				}
+			}
+
+			return nil
+		}, 5*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+		ginkgo.By("Removing the first peer")
+		peer0 := peers[0]
+		peer0.Namespace = ConfigUpdater.Namespace()
+		err = ConfigUpdater.Client().Delete(context.TODO(), &peer0)
+		Expect(err).ToNot(HaveOccurred())
+
+		expectedPeers := peersNames[1:]
+		expectDeletion := len(peers) == 1
```

Comment:
why that? when we have that true?

Reply:
the `HostContainerSetup` scenario (when we use one host networked container)